### PR TITLE
Add system demo examples

### PR DESCRIPTION
**Pull request by rchan26**

### CODE REVIEW

**Review by edchapman88 — CHANGES_REQUESTED** [review]
Looks great! Found a typo (hence the `request changes`), left a few re-wording suggestions, and a couple of very optional suggestions (cutting a large section from experiment 3, and moving the naive sync send function to be visible in the notebook code). So cool to see the numbers on the page about how fast it is, great work!

File: examples/system-demo/experiment_3.ipynb [hunk 1]
```diff
@@ -0,0 +1,731 @@
...
+    "We can also run the above experiments via the command line. The command is as follows (assuming that your working directory is the current directory of this notebook, i.e. `examples/system-demo`):\n",
+    "```bash\n",
+    "prompto_run_experiment --file data/input/openai-gpt-3pt5-turbo.jsonl --max_queries 500\n",
+    "prompto_run_experiment --file data/input/openai-gpt-4.jsonl --max_queries 500\n",
+    "prompto_run_experiment --file data/input/openai-gpt-4o.jsonl --max_queries 500\n",
+    "```\n",
+    "\n",
+    "But for this notebook, we will time the experiments and save them to the `sync_times` and `prompto_times` dictionaries."
+   ]
+  },
```
[the remark this answers is not in the document]
Comment (rchan26) [reply examples/system-demo/experiment_3.ipynb:492]: good point. we do use these results in the paper though, I'll swap the orders around

File: examples/system-demo/experiment_3.ipynb [hunk 2]
```diff
@@ -0,0 +1,731 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice here that we are setting `parallel=True` in the `Settings` object as well as specifying the rate limits to send to each of the models, and we specify each of them to be 500.  We do this by passing in a dictionary to the `max_queries_dict` argument in the `Settings` object which has API names as the keys and the values are also a dictionary where the keys are the model names we wish to process in parallel and the values are rate limits.\n",
+    "\n",
+    "For details of how to specify rate limits for _different models in the same API_, see the [Specifying rate limits docs](../../docs/rate_limits.md) and the [Grouping prompts and specifying rate limits notebook](../notebooks/grouping_prompts_and_specifying_rate_limits.ipynb).\n",
+    "\n",
+    "Note that in the [previous experiment](./experiment_2.ipynb), we also used parallel processing but in a slightly different way as we were parallelising the querying of different APIs."
```
[the remark this answers is not in the document]
Comment (rchan26) [reply examples/system-demo/experiment_3.ipynb:579]: I'll make it a header to signify this more

File: examples/system-demo/api_utils.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+)
+from ollama import Client
+from openai import OpenAI
+from openai.types.chat import ChatCompletion
+from tqdm import tqdm
+
+
+def send_prompts_sync(prompt_dicts: list[dict]) -> list[str]:
+    # maive for loop to synchronously dispatch prompts
+    return [send_prompt(prompt_dict) for prompt_dict in tqdm(prompt_dicts)]
```
[the remark this answers is not in the document]
Comment (rchan26) [reply examples/system-demo/api_utils.py]: yeah I think this is a good idea, duplication isn't too big of a problem I guess in this case

**Pull request merged**